biomage/utils/data.py

```diff
@@ -1,33 +1,95 @@
 import boto3
 import click
+from biomage.experiment.utils import (add_env_user_to_experiment,
+                                      create_gem2s_hash,
+                                      get_experiment_project_id)
 from botocore.exceptions import ClientError
+from click.utils import echo
 
 from ..utils.constants import PRODUCTION, STAGING
 
 
-def modified_records(item, target_table, config):
+def remap_sample_references(samples, sandbox_id):
+    """
+    Edit entries in samples dictionary to opoint to the right prefixed resources
+    """
+
+    remapped_samples = {"M" : {}}
+
+    for sample_id in samples['M']:
+
+        prefixed_sample_name = f"{sandbox_id}-{sample_id}"
+        prefixed_project_uuid = f"{sandbox_id}-{samples['M'][sample_id]['M']['projectUuid']['S']}"
+
+        remapped_samples['M'][prefixed_sample_name] = {
+                "M": {
+                    **samples['M'][sample_id]['M'],
+                    "files": remap_file_references(
+                        samples['M'][sample_id]["M"]["files"],
+                        sandbox_id
+                    ),
+                    "uuid": {"S" : prefixed_sample_name},
+                    "projectUuid": {
+                        "S" : prefixed_project_uuid
+                    }
+                }
+        }
+
+    return remapped_samples
+
+
+def remap_file_references(files, sandbox_id):
+    """
+    Edit entries in files dictionary to opoint to the right prefixed resources
+    """
+
+    valid_filenames = [file for file in files["M"] if file != "lastModified"]
+    
+    for file in valid_filenames:
+        files['M'][file]['M']['path']['S'] = f"{sandbox_id}-{files['M'][file]['M']['path']['S']}"
+
+    return files
+
+
+def modify_records(item, target_table, config, **extra):
     """
     Return modified records.
     This function should return spreadable dictionary
     """
 
     if target_table == config["staging-experiments-table"]:
+
+        item['projectId']['S'] = f"{extra['sandbox_id']}-{item['projectId']['S']}"
+
+        new_rbac = add_env_user_to_experiment(cfg=item)['rbac_can_write']
+        item['rbac_can_write'] = new_rbac
```

Comment:
this is just `cfg = add_env_user_to_experiment(cfg=item)`

Reply:
Added